gdnative-core/src/geom/basis.rs

```diff
@@ -73,42 +116,39 @@ impl Basis {
         }
     }
 
-    /* FIXME: Sanity tests fail
     /// Inverts the matrix.
     pub fn invert(&mut self) {
+        let [x, y, z] = self.elements;
+
         let co = [
-            self.elements[1].y * self.elements[2].z - self.elements[1].z * self.elements[2].y,
-            self.elements[1].z * self.elements[2].x - self.elements[1].x * self.elements[2].z,
-            self.elements[1].x * self.elements[2].y - self.elements[1].y * self.elements[2].x,
+            y.y * z.z - y.z * z.y,
+            y.z * z.x - y.x * z.z,
+            y.x * z.y - y.y * z.x,
         ];
 
-        let det: f32 =
-            self.elements[0].x * co[0] + self.elements[0].y * co[1] + self.elements[0].z * co[2];
-        debug_assert!(!det.approx_eq(&0.0), "Determinant was zero");
+        let det: f32 = x.x * co[0] + x.y * co[1] + x.z * co[2];
+        assert!(!det.approx_eq(&0.0), "Determinant was zero");
 
         let s: f32 = 1.0 / det;
 
         self.set_x(Vector3::new(co[0] * s, co[1] * s, co[2] * s));
-
         self.set_y(Vector3::new(
-            (self.elements[0].z * self.elements[2].y - self.elements[0].y * self.elements[2].z) * s,
-            (self.elements[0].x * self.elements[2].z - self.elements[0].z * self.elements[2].x) * s,
-            (self.elements[0].y * self.elements[2].x - self.elements[0].x * self.elements[2].y) * s,
+            (x.z * z.y - x.y * z.z) * s,
+            (x.x * z.z - x.z * z.x) * s,
+            (x.y * z.x - x.x * z.y) * s,
         ));
-
         self.set_z(Vector3::new(
-            (self.elements[0].y * self.elements[1].z - self.elements[0].z * self.elements[1].y) * s,
-            (self.elements[0].z * self.elements[1].x - self.elements[0].x * self.elements[1].z) * s,
-            (self.elements[0].x * self.elements[1].y - self.elements[0].y * self.elements[1].x) * s,
+            (x.y * y.z - x.z * y.y) * s,
+            (x.z * y.x - x.x * y.z) * s,
+            (x.x * y.y - x.y * y.x) * s,
         ));
     }
 
     /// Returns the inverse of the matrix.
-    pub fn inverse(mut self) -> Basis {
+    pub fn into_inverse(mut self) -> Basis {
```

Comment:
I see that CI is passing for `invert`. Great! I'm afraid that this name change is a [hypercorrection](https://en.wikipedia.org/wiki/Hypercorrection), though. In the Rust ecosystem, `to_` and `into_` are only used for type conversion methods. `f64::recip`, for example, is named `recip` instead of `into_recip`. One doesn't add `into_` *just* because it takes `self`.

Reply:
I'll correct the `into_` renames. IIRC, there was a situation where 3 functions were `scale`, `scale` and `scaled`.
Where the first `scale` represented getting the current scale, the second would scale the matrix, and the third would consume and scale the matrix. Do you have a preference on disambiguating the first 2?

Reply:
Oh, and I'll squash the PR.

Reply:
I left `quat` and `into_quat` as I didn't have a past tense `quat`. 
Squash is done.

Reply:
On `quat` and `into_quat`: I don't think they should both exist? Seems like `into_quat` just skips the part that "ensures" that `self` is a rotation matrix, but then it calculates `determinant` and `is_orthogonal` again so I'm not sure if there is any gain to be had. I'd suggest merging the contents of `into_quat` into `quat`.